src/anndata/_io/specs/methods.py

```diff
@@ -1216,12 +1206,22 @@ def _string_array(
     read_nullable_boolean
 )
 
-_REGISTRY.register_read(H5Group, IOSpec("nullable-string-array", "0.1.0"))(
-    read_nullable_string := partial(_read_nullable, array_type=_string_array)
-)
-_REGISTRY.register_read(ZarrGroup, IOSpec("nullable-string-array", "0.1.0"))(
-    read_nullable_string
-)
+
+@_REGISTRY.register_read(H5Group, IOSpec("nullable-string-array", "0.1.0"))
+@_REGISTRY.register_read(ZarrGroup, IOSpec("nullable-string-array", "0.1.0"))
+def _read_nullable_string(
+    elem: GroupStorageType, *, _reader: Reader
+) -> pd.api.extensions.ExtensionArray:
+    values = _reader.read_elem(elem["values"])
+    mask = _reader.read_elem(elem["mask"])
+    dtype = pd.StringDtype()
+
+    arr = pd.array(
+        values.astype(NULLABLE_NUMPY_STRING_TYPE),
```

Comment:
I think this might be causing the current test failures.

I didn’t really build the PR with numpy 1 in mind, so backporting it might be slightly involved due to that additional degree of freedom.

Reply:
This doesn't have an impact on the PR - the failures were just related to my turning on `infer_string` for pre-release and forgetting everythign stops roundtripping at that point - I added an explicit test for `infer_string` now